### DISCUSSION TITLE
error: Failed to retrieve container image info: Error: No such image

### DISCUSSION THREAD

**Discussion by bertdeelman** [containrrr/watchtower#1036] [General] [answered]
Hi,

I use Watchttower at a Synology NAS. I use the default configuration. After starting the container I receive two (errors/warming) 

- Failed to retrieve container image info: Error: No such image: sha256:.....
- watchtower cron: panic running job: runtime error: invalid memory address or nil pointer dereference

Any idea what to do?

Regards,
Bert

**Comment by piksel** [comment]
Are you using `docker compose` by any chance? Not `docker-compose`, but the "v2" integrated command. It doesn't store the image name in a retrievable manner and does not work with watchtower unfortunately.

**Reply by bertdeelman** [reply]
Hi,

I don't quite understand what you mean. 

I tried both. So from the command line: 
`$ docker run -d \
--name watchtower \
-v /var/run/docker.sock:/var/run/docker.sock \
containrrr/watchtower`

And a with docker-compose.yaml file, with the `docker-compose` command. Same notifications.

**Reply by piksel** [reply]
I mean for the other containers on your system.
If you run `docker ps`, does all the containers have a name in the `IMAGE` column?
If it's instead a hash, it causes the above error in watchtower (and is caused by `docker compose` as far as we know).

#1027 addresses this, but only skips containers like that, it still cannot check/update them.

**Comment by bertdeelman** [comment]
ok, they have a name not a hash like the error message. I've got some strange image names like: 7c6d73bed103 or c060f9cb7fd5, but most of them are normal like bastilimbach/docker-magicmirror:latest etc.

**Reply by piksel** [reply]
Yeah, those are hashes 😁 (its the "short" format).

`containrrr/watchtower:latest-dev` contains the patch that should allow watchtower to skip those instead of crashing, but it's pre-release so it could be unstable. If you have any idea how you ended up with those containers it could be helpful.

**Comment by bertdeelman** [comment, marked as answer]
Hi,

I updated all images with a hash in their names by hand and use the `latest-dev` version of Watchtower. And for now it has successfully upgrade containers. 

Thanks!

Regards,
Bert

**Reply by piksel** [reply]
Nice! We haven't heard from docker whether those hash-only containers are supposed to not contain the image name or not. If that is the new format, watchtower will be severely crippled as we rely on that field... 

Glad that it works for you now at least!

**Reply by billouetaudrey** [reply]
Hi, How you update all image please?

**Reply by piksel** [reply]
@billouetaudrey I don't see how this question relates to this answer. Create a new discussion with your question and more details.

**Reply by billouetaudrey** [reply]
@piksel I have the same issue " Error: failed to retrieve container ... " and he found the fix so I ask how he update all images ;) It's not off topic

**Reply by piksel** [reply]
@billouetaudrey Fair enough. I think he simply means removing the containers and re-creating them again using `docker run` or `docker-compose up`. You cannot change the image of a created container.

**Reply by billouetaudrey** [reply]
Ok thanks for reply :)
Good luck :)

**Comment by LostOnTheLine** [comment]
I just ran into this issue, but the comments here helped me figure out how to fix it.

My issue involved a container having a **"hash"** name, but it was the result, I believe, of an error, or perhaps the server froze or restarted during a watchtower update.
I went into Portainer & found the container, then stopped it's stack & restarted the stack. After I did that I stopped getting the error as it now used the updated image for that container instead of the old one without a name.

The `Docker PS` command was helpful, but if you use Portainer going to the Containers List would be more helpful, especially if you have it set to update containers that are not currently running

**Reply by LostOnTheLine** [reply]
Just verified that just deleting any containers that have a hash name fixes this as well as I just saw it for a container that was currently stopped & just deleting it worked.

**Comment by sameleff** [comment]
I am experiencing this same error, but none of my containers have hashed names according to the `docker ps` command. Are there any other causes of this behavior?
